### ISSUE TITLE
[OSX] 1.7.7 crashes - unable to run any N64 games (bivert included)

### ISSUE THREAD

**Issue by gingerbeardman** [libretro/RetroArch#8744]
## Description

1.7.7 crashes - unable to load any games at all (1.7.5 works)

### Expected behavior

- Should work, as 1.7.6 did

### Actual behavior

- Crashes, hard [crash dump](https://gist.githubusercontent.com/gingerbeardman/204c3d89db95c0f815841dc9439e2230/raw/f4bc9f91927973f2fdb5be14c1d53b2688f18bda/gistfile1.txt)

### Steps to reproduce the bug

1. Download 1.7.7 macOS Metal
2. load content, run

### Bisect Results

For N64 on mupen64plus or Parallel64:

- 1.7.5 Metal _works_
- 1.7.6 (24 March 2019 nightly?) Metal _does not work_
  (there is no working Stable 1.7.6 Metal download)

### Version/Commit
1.7.7 stable

### Environment information

- OS: macOS 10.13.6

**Comment by bparker06** [comment]
Metal only works with software-rendered cores. N64 cores typically require OpenGL, unless you use the software-rendered (and highly demanding) angrylion renderer in parallel.

**Comment by orbea** [comment]
Is there anything that can be done here? Without adding metal support to the cores or fixing opengl in osx I suspect there is not much hope.

**Comment by bparker06** [comment]
What needs fixing? There is already a non-Metal build that supports OpenGL.

**Comment by orbea** [comment]
I presume OpenGL support is just going to get worse and worse in osx as time goes on...

**Comment by RobLoach** [comment]
:penguin:

Does MacOS not work with Vulkan either? Wow... MoltenVK may be an option?

**Comment by orbea** [comment]
> MoltenVK may be an option?

If there is someone willing and capable of maintaining it, I don't think we have many developers using osx in general.

**Comment by twinaphex** [comment]
This is not really a bug - I believe we took out OpenGL out of the Metal version since it seems OpenGL and Metal can't seem to peacefully co-exist - tons of weird bugs and issues would happen if we attempted to mix both.

I honestly don't know how to fix it - neither can @stuartcarnie or @cold-brewed . So the decision was made to just make RetroArch Metal build 'Metal-only' and have a separate version for OpenGL.

OpenGL is on the way out anyway as far as Apple is concerned - they're already breaking things in their SDK causing some display issues with GL apparently if you build it with an SDK newer than 10.11. The Metal-only version is at least future proof for when Apple removes it altogether.

**Comment by gingerbeardman** [comment]
Thanks for the info. 

I had read in the 1.7.7 blog post about Metal progress, but I was not aware that means some cores are essentially deprecated going forward with macOS. 

So I guess there needs to be more visible indication for users about this, given that hands are tied. I imagine bug reports like this one will keep coming?

Maybe the different macOS versions can be labelled in such a way as to show compatibility with cores?

On my 10.13.6 setup I don't mind rolling back to 1.7.5 for Metal + GL until I figure out what is the best approach. 

I'll test the current GL build shortly. 

Ideally users like me would just follow your recommendation if there is one?

**Comment by gingerbeardman** [comment]
non-Metal builds have their own problems, eg. https://github.com/libretro/RetroArch/issues/8761

**Comment by gingerbeardman** [comment]
"completed" in what way? @warmenhoven

**Comment by warmenhoven** [comment]
The Metal build now provides metal, gl, glcore, and Vulkan graphics drivers; the mupen64 core is able to use gl/glcore/vulkan without problems. There are no plans to add a Metal renderer to libretro or any core.

**Comment by gingerbeardman** [comment]
Awesome!
